### PR TITLE
CN-2206: multipart file handling example

### PR DESCRIPTION
**Pull request by johnbastian-trayio**
- added `file-handling-example` connector that uses dropbox API for regular uploading and downloading of a file
- added `multipart-file-handling-example` connector that uses imgur API for multipart uploading with a file

### CODE REVIEW

File: multipart-file-handling-example/src/download_image/handler.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	(handlerTest) =>
+		handlerTest
+			.usingHandlerContext('test')
+			.nothingBeforeAll()
+			.testCase('should download image', (testCase) =>
+				testCase
+					.givenNothing()
+					.when(() => ({ image_hash: 'CKVpBov' }))
+					.then(({ output }) => {
+						console.log(output);
```
Comment (gnyani-trayio) [inline multipart-file-handling-example/src/download_image/handler.test.ts:17]: drop this?

**Review by gnyani-trayio — DISMISSED** [review]

File: file-handling-example/connector.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+	"name": "file-handling-example",
+	"version": "1.0",
+	"title": "File handling example",
+	"description": "",
+	"service": {
+		"name": "json-placeholder",
+		"version": "1"
+	},
```
Comment (joaoTray) [inline file-handling-example/connector.json:9]: do we need to change this name to match the name of the connector?
is this field related to service at tray?

File: file-handling-example/src/download_file/handler.test.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+OperationHandlerTestSetup.configureHandlerTest(
+	downloadFileHandler,
+	(handlerTest) =>
+		handlerTest
+			.usingHandlerContext("test")
+			.nothingBeforeAll()
+			.testCase("should download a file", (testCase) =>
+				testCase
+					.givenNothing()
+					.when(() => ({ file_path: '/cdk-testing/meme.jpeg' }))
```
Comment (joaoTray) [inline file-handling-example/src/download_file/handler.test.ts:15]: can we point to a file that exists?
people that will download the example won't notice that this file is not part of the repo?

File: file-handling-example/src/download_file/handler.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+export const downloadFileHandler = OperationHandlerSetup.configureHandler<
+	FileHandlingExampleAuth,
+	DownloadFileInput,
+	DownloadFileOutput
+	>((handler) =>
+		handler.usingHttp((http) =>
+			http
+				.get(
+					"https://content.dropboxapi.com/2/files/download"
```
Comment (joaoTray) [inline file-handling-example/src/download_file/handler.ts:15]: should we maybe suggest to create a server in postman to use as a "downloable" server?

File: file-handling-example/src/upload_file/handler.test.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+			.usingHandlerContext("test")
+			.nothingBeforeAll()
+			.testCase("should upload file", (testCase) =>
+				testCase
+					.givenNothing()
+					.when(() => ({
+						file_path: '/cdk-testing/meme.jpeg',
+						file: {
+							name: "meme.jpg",
+							url: "https://i.imgur.com/CKVpBov.jpg",
```
Comment (joaoTray) [inline file-handling-example/src/upload_file/handler.test.ts:19]: why we have a name and a url with different file names?
should the url not only contain the url without the filename?
i don't understand this example, we have file_path and file. Not sure what this means? Maybe a comment would help?

File: file-handling-example/src/download_file/handler.ts (hunk 4 above)
Comment (joaoTray) [reply file-handling-example/src/download_file/handler.ts:15]: ok, all is dropbox. Let's leave as is. 👍

File: multipart-file-handling-example/connector.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+	"name": "multipart-file-handling-example",
+	"version": "1.0",
+	"title": "Multipart file handling example",
+	"description": "",
+	"service": {
+		"name": "json-placeholder",
+		"version": "1"
+	},
```
Comment (joaoTray) [inline multipart-file-handling-example/connector.json:9]: same question about this field: service

File: multipart-file-handling-example/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+		"strict": true,
+		"noImplicitAny": true, 
+		"strictNullChecks": true,                       
+		"noImplicitThis": true,            
+		"useUnknownInCatchVariables": true,
+      "skipLibCheck": true,
+      "rootDir": "./src",
+      "outDir": "./dist"
+  }
+}
```
Comment (joaoTray) [inline multipart-file-handling-example/tsconfig.json:16]: formatting

File: file-handling-example/src/download_file/handler.test.ts (hunk 3 above)
Comment (johnbastian-trayio) [reply file-handling-example/src/download_file/handler.test.ts:15]: This is the file path in dropbox itself, not something local.

And if they run `upload_file` test first, then the file will exist

File: file-handling-example/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+		"strict": true,
+		"noImplicitAny": true, 
+		"strictNullChecks": true,                       
+		"noImplicitThis": true,            
+		"useUnknownInCatchVariables": true,
+      "skipLibCheck": true,
+      "rootDir": "./src",
+      "outDir": "./dist"
+  }
+}
```
[the remark this answers is not in the document]
Comment (johnbastian-trayio) [reply file-handling-example/tsconfig.json:16]: I didn't generate the file 🤔

File: file-handling-example/src/upload_file/handler.test.ts (hunk 5 above)
Comment (johnbastian-trayio) [reply file-handling-example/src/upload_file/handler.test.ts:19]: Go to that URL; it's an actual meme 😛

The `file` is a FileReference object - the thing passed around in workflows. 
The `file_path` is one of the arguments dropbox requires - it's where to save the file

Comment (johnbastian-trayio) [reply file-handling-example/src/upload_file/handler.test.ts:19]: Also, typically the URL in the `file`/FileReference object will be a signed S3 URL, if it's coming from a workflow.

File: file-handling-example/src/download_file/handler.test.ts (hunk 3 above)
Comment (joaoTray) [reply file-handling-example/src/download_file/handler.test.ts:15]: Maybe add a comment then? Because I didn't know this before hand and this is an example repo?

File: file-handling-example/src/upload_file/handler.test.ts (hunk 5 above)
Comment (joaoTray) [reply file-handling-example/src/upload_file/handler.test.ts:19]: maybe worth a comment then? (again, only because is an example repo?)

File: file-handling-example/tsconfig.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (joaoTray) [reply file-handling-example/tsconfig.json:16]: ok

File: multipart-file-handling-example/tsconfig.json (hunk 7 above)
Comment (joaoTray) [reply multipart-file-handling-example/tsconfig.json:16]: auto-generated

**Review by joaoTray — APPROVED** [review]
